Sources/Plasma/FeatureLib/pfPasswordStore/pfPasswordStore_Apple.cpp

```diff
@@ -80,13 +80,37 @@ ST::string pfApplePasswordStore::GetPassword(const ST::string& username)
 bool pfApplePasswordStore::SetPassword(const ST::string& username, const ST::string& password)
 {
     ST::string service = GetServerDisplayName();
+    
+    OSStatus err = SecKeychainAddGenericPassword(nullptr,
+                                                service.size(),
+                                                service.c_str(),
+                                                username.size(),
+                                                username.c_str(),
+                                                password.size(),
+                                                password.c_str(),
+                                                nullptr);
+    if(err==errSecDuplicateItem) {
+        //the keychain item already exists, update it
+        CFStringRef cfUsername = CFStringCreateWithCString(NULL, username.c_str(), kCFStringEncodingUTF8);
+        CFDataRef cfPassword = CFDataCreate(NULL, (const UInt8*)password.c_str(), password.size());
+        CFStringRef cfServiceName = CFStringCreateWithCString(NULL, service.c_str(), kCFStringEncodingUTF8);
+        CFAutorelease(cfUsername);
+        CFAutorelease(cfPassword);
+        CFAutorelease(cfServiceName);
+        
+        const void *queryKeys[2] = { kSecClass, kSecAttrService };
+        const void *queryValues[2] = { kSecClassGenericPassword, cfServiceName };
```

Comment:
```suggestion
        const void* queryKeys[2] = { kSecClass, kSecAttrService };
        const void* queryValues[2] = { kSecClassGenericPassword, cfServiceName };
```

Pointers bind to the type, not the variable name.